### PR TITLE
OCPNODE-2226: Introduce versioning in auto-node-sizing feature

### PR DESCRIPTION
**Pull request by sairameshv**
**- What I did**
Introducing versioning for the auto-node-sizing feature
**- How to verify it**
The presence of `/etc/node-sizing-version.json` file on a node indicates the introduction of versioning in the auto node sizing feature
**- Description for the changelog**
Auto node sizing feature gives recommendations about the `kube reserved`, `system reserved` memory and cpus as described in this [enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/kubelet/kubelet-node-sizing.md)
These recommendations tend to change on a requirement basis i.e. based on the resource utilizations of system and kube, new features etc.
Hence, introduction of the versioning for this feature helps in modifying the memory, cpu values and also ease the upgrades without breaking the clusters.

### CODE REVIEW

File: templates/common/_base/files/kubelet-auto-sizing.yaml [hunk 1]
```diff
@@ -113,12 +132,17 @@ contents:
...
         set_es $3
     }
-
+    func create_version_file {
+        echo "{\"version\": $1}" > $2
+    }
+    if ! [ -f $NODE_AUTO_SIZING_VERSION_FILE ]; then
+        create_version_file $NODE_AUTO_SIZING_VERSION $NODE_AUTO_SIZING_VERSION_FILE
+    fi
+    new_version=`jq .version $NODE_AUTO_SIZING_VERSION_FILE`
```
Comment (rphillips) [inline templates/common/_base/files/kubelet-auto-sizing.yaml:141]: should be: ```$(jq .version $NODE_AUTO_SIZING_VERSION_FILE)```

**Review by sinnykumari — COMMENTED** [review]
Looks sane
/approve
Putting hold in case we want to get QE pre-merge testing. If node team follows post merge testing for feature work, feel free to remove hold.
/hold

**Comment by rphillips** [comment]
/hold cancel

I don't believe we need a pre-merge QE test on this.

**Pull request merged**